packages/core/saved-objects/core-saved-objects-server/src/security.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObject } from '@kbn/core-saved-objects-common';
+import { EcsEventOutcome } from '@kbn/logging';
+
+/**
+ * The CheckAuthorizationParams interface contains settings for checking
+ * authorization via the ISavedObjectsSecurityExtension.
+ */
+export interface CheckAuthorizationParams<A extends string> {
+  /**
+   * A set of types to check.
+   */
+  types: Set<string>;
+  /**
+   * A set of spaces to check.
+   */
+  spaces: Set<string>;
+  /**
+   * An array of actions to check.
+   */
+  actions: A[];
```

Comment:
Speculating here: Perhaps we allow duplicate actions but spaces and types have to be unique?

Reply:
There is no need for duplicate actions, but I think this came down to simplicity. Types and spaces are generated iteratively and there is a need to ensure uniqueness. The actions array is passed in statically in all calls, so it would be quite deliberate if it had a duplicate in it. 

I am happy to change the actions array to a Set in order to ensure no duplicates if you all agree. The only downside I could imagine is if Sets are particularly slow of memory hungry (then again we're only talking about a one item Set 100% of the time right now).